src/drivers/uavcan_v1/templates/_composite_type.j2

```diff
@@ -0,0 +1,115 @@
+{{ composite_type | definition_begin }}
+{
+{%- if T.fixed_port_id is not None %}
+	static constexpr CanardPortID PORT_ID = {{T.fixed_port_id}};
+{%- endif %}
+
+{%- set total_size = namespace(value=0) -%}
+{%- for field in composite_type.fields -%}
+    {% set total_size.value = total_size.value + field.data_type.bit_length_set|max %}
+{%- endfor %}
+	static constexpr size_t SIZE = {{total_size.value}};
+
+	static constexpr auto getDataTypeFullName() { return "{{T}}"; }
```

Comment:
This might bloat up your binary with a lot of text depending on how many types you use. How much flash do you have budgeted?